Changed api to use builtin python types instead of JSON

NickolausDS opened the pull request:
Changes proposed in #2. When calling into the api, it's now possible to do: 

```
from identifiers_client import config
from identifiers_client import identifiers_api
ic = identifiers_api.identifiers_client(config)

ic.create_identifier(visible_to=['public'], namespace='<namespace>')
ic.update_identifier(<identifier>, location=['foo.example.com'])
```
This changes also preserve the way CLI args are specified:
```
$ globus-identifiers-client identifier-create --namespace <namespace> --visible-to '["public"]'
```

bjmc-globus commented:
I think this looks fine, but we should be careful about changing the public interface for the CLI. Do we know how many groups are using it? At a minimum, we should bump the version if we're make a breaking change.

NickolausDS commented:
Removed `[WIP]` and rebased the changes so these are based on `v0.2.0` and not `v0.1.0`.

From talking to Jim, I'll not include changes in #2 for the CLI, and only include the current changes to the SDK. That way the PR doesn't get too large.

NickolausDS commented:
I also went over creating an identifier again just as a quick sanity check after rebasing. Everything looks good.
